akka-cluster-typed/src/main/scala/akka/cluster/ddata/typed/javadsl/ReplicatorMessageAdapter.scala

```diff
@@ -64,7 +64,7 @@ class ReplicatorMessageAdapter[A, B <: ReplicatedData](
    * the replicator are transformed to the message protocol of the requesting actor with
    * the given `responseAdapter` function.
    */
-  def subscribe(key: Key[B], responseAdapter: JFunction[Replicator.SubscribeResponse[B], A]): Unit = {
```

Comment:
JFunction seems to be used in more places here

Reply:
Here `JFunction` is actually `java.util.Function`, so maybe we should align those to be only japi and call the java.util something else when we rename-import it?

Reply:
I think the response adaptation lambdas here should be the japi one though.

Reply:
I take that back, the message adaption lambda here is always successful response -> message so I think that should be pure and not throw.